presto-main/src/main/java/com/facebook/presto/operator/PagesIndex.java

```diff
@@ -437,6 +437,16 @@ public LookupSourceSupplier createLookupSourceSupplier(
         return createLookupSourceSupplier(session, joinChannels, hashChannel, filterFunctionFactory, sortChannel, searchFunctionFactories, Optional.empty());
     }
 
+    public PagesSpatialIndex createPagesSpatialIndex(
+            Session session,
+            int geometryChannel,
+            Optional<JoinFilterFunctionFactory> filterFunctionFactory,
+            List<Integer> outputChannels)
+    {
+        List<List<Block>> channels = ImmutableList.copyOf(this.channels);
```

Comment:
I don't know. I followed `PagesIndex#createLookupSourceSupplier` which makes a copy of `channels`. Is there any reason these two would be different in that regard?

Reply:
Interesting. I don't know why `PagesIndex#createLookupSourceSupplier` copies `channels`.. Maybe it's a (seemingly failed) attempt to make created `LookupSourceSupplier` independent from `PagesIndex`. Failed, because `valueAddresses` are passed as-is.

I think not copying  is good from memory _accounting_ stand-point. Once you created the spatial index, you ask it how much memory it uses. If you don't share objects with `PagesIndex`, you need to add memory used by `PagesIndex` too... if you share only portion of state with `PagesIndex`, then things get more complicated.

Reply:
You can leave as-is (doing the same as in `PagesIndex#createLookupSourceSupplier` has its own merits), but please leave `// TODO probably shouldn't copy to reduce memory and for memory accounting's sake`